src/GafferScene/OptionQuery.cpp

```diff
@@ -291,60 +291,47 @@ void OptionQuery::compute( Gaffer::ValuePlug *output, const Gaffer::Context *con
 
 		if( output == oPlug->getChild( g_existsPlugIndex ) )
 		{
-			const NameValuePlug *childQueryPlug = queryPlug( output );
-
-			const std::string optionName = childQueryPlug->namePlug()->getValue();
-
+			const std::string optionName = queryPlug( output )->namePlug()->getValue();
+			bool exists = false;
 			if( optionName.size() )
 			{
 				ConstCompoundObjectPtr globals = scenePlug()->globalsPlug()->getValue();
-				const Data *resultData = globals->member<const Data>( g_namePrefix + optionName );
-
-				static_cast<BoolPlug *>( output )->setValue( resultData != nullptr );
-
-				return;
+				exists = globals->members().count( g_namePrefix + optionName );
 			}
-
-			static_cast<BoolPlug *>( output )->setValue( false );
-
+			static_cast<BoolPlug *>( output )->setValue( exists );
 			return;
 		}
 
-		else if(
-			oPlug->getChild( g_valuePlugIndex )->isAncestorOf( output ) ||
-			output == oPlug->getChild( g_valuePlugIndex )
-		)
+		const ValuePlug *valuePlug = oPlug->getChild<ValuePlug>( g_valuePlugIndex );
+		if( output == valuePlug || valuePlug->isAncestorOf( output ) )
 		{
 			const NameValuePlug *childQueryPlug = queryPlug( output );
 
 			const std::string optionName = childQueryPlug->namePlug()->getValue();
-
-			const ValuePlug *vPlug = valuePlugFromQuery( childQueryPlug );
-
+			ConstObjectPtr object;
 			if( optionName.size() )
 			{
 				ConstCompoundObjectPtr globals = scenePlug()->globalsPlug()->getValue();
-				const Data *resultData = globals->member<const Data>( g_namePrefix + optionName );
+				object = globals->member<Object>( g_namePrefix + optionName );
+			}
 
-				if( resultData != nullptr )
+			if( object )
+			{
+				if( auto objectPlug = runTimeCast<ObjectPlug>( output ) )
+				{
+					objectPlug->setValue( object );
+					return;
+				}
+				else if( auto data = runTimeCast<const Data>( object.get() ) )
 				{
-					if( PlugAlgo::setValueFromData( vPlug, output, resultData ) )
-					{
-						return;
-					}
+					PlugAlgo::setValueFromData( valuePlug, output, data );
+					return;
```

Comment:
If we fail to set the data in the line above, say because the option type is a string and the query type is a float, this will return without setting the plug value and I get an error like `Gaffer.ProcessException: OptionQuery.out.out0.value : Compute did not set plug value.`.

Before, we were setting it to the default value, is that preferable? Or do we want it to error to alert the user that their query is broken?

Reply:
Thanks for spotting that - it was completely unintentional. I'll sort it out.